sendlib.c

```diff
@@ -806,20 +805,27 @@ static size_t convert_file_to(FILE *fp, const char *fromcode, int ncodes,
   if (cd1 == (iconv_t)(-1))
     return -1;
 
-  iconv_t *cd = mutt_mem_calloc(ncodes, sizeof(iconv_t));
-  size_t *score = mutt_mem_calloc(ncodes, sizeof(size_t));
-  struct ContentState *states = mutt_mem_calloc(ncodes, sizeof(struct ContentState));
-  struct Content *infos = mutt_mem_calloc(ncodes, sizeof(struct Content));
+  iconv_t *cd = mutt_mem_calloc(tocodes->count, sizeof(iconv_t));
+  size_t *score = mutt_mem_calloc(tocodes->count, sizeof(size_t));
+  struct ContentState *states =
+      mutt_mem_calloc(tocodes->count, sizeof(struct ContentState));
+  struct Content *infos = mutt_mem_calloc(tocodes->count, sizeof(struct Content));
 
-  for (int i = 0; i < ncodes; i++)
   {
-    if (mutt_str_strcasecmp(tocodes[i], "utf-8") != 0)
-      cd[i] = mutt_ch_iconv_open(tocodes[i], "utf-8", 0);
-    else
+    int i = 0;
```

Comment:
I decided to add another "scope" here so the counter (`i`) would go out of scope after this block of code as it is a very frequently used identifier and kinda intuitive when you see it int he code. It doesn't have a use after that loop.

It is probably not the best solution but without a `STAILQ_FOREACH_WITH_COUNTER` (or similar) macro that seemed reasonable.